### PR TITLE
feat: currentVersionAgeInDays, newVersionAgeInDays

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Adds `currentVersionAgeInDays` dependencies if `currentReleaseTimestamp` is known. Also adds `newVersionAgeInDays` to an update if the update has a `releaseTimestamp`.

## Context

For use in `matchJsonata`

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/process/lookup/index.ts [hunk 1]
```diff
@@ -312,9 +312,16 @@ export async function lookupUpdates(
 
       if (is.nonEmptyString(currentVersionTimestamp)) {
         res.currentVersionTimestamp = currentVersionTimestamp;
+        // Calculate age in days
+        const currentVersionTimestampDate = new Date(currentVersionTimestamp);
+        const now = new Date();
+        const diffInMs = now.getTime() - currentVersionTimestampDate.getTime();
+        const currentVersionAge = Math.ceil(diffInMs / (1000 * 60 * 60 * 24));
```
Comment (viceice) [inline lib/workers/repository/process/lookup/index.ts:319]: use luxon, will simplify the code and is much more safe for parsing errors

File: lib/util/date.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import { DateTime } from 'luxon';
 
 const ONE_MINUTE_MS = 60 * 1000;
-const ONE_DAY_MS = 24 * 60 * ONE_MINUTE_MS;
 
 export function getElapsedDays(timestamp: string): number {
```
Comment (rarkins) [inline lib/util/date.ts:5]: This function was previously unused

**Comment by rarkins** [comment]
I think these fields could enable a rule like this:

```
{
  "description": "Don't update renovate more than weekly",
  "matchJsonata": "packageName = 'renovate/renovate' and newVersionAgeInDays - currentVersionAgeInDays < 7",
  "enabled": false
}
```

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/util/date.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-  return Math.floor(
-    (new Date().getTime() - new Date(timestamp).getTime()) / ONE_DAY_MS,
-  );
+  const currentVersionTimestampDate = DateTime.fromISO(timestamp);
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/util/date.ts:6]: In theory they're all validated before they reach this point, so I think we can leave this as-is unless a problem is found

**Review by secustor — APPROVED** [review]

**Pull request merged**